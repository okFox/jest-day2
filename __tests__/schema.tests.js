const Schema = require('../lib/Schema');

describe('Schema', () => {
  it('validates a good schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      color: {
        type: String
      }
    });

    const cat = {
      name: 'whiskey',
      age: 18,
      color: 'grey'
    };

    expect(schema.validate(cat)).toEqual({
      name: 'whiskey',
      age: 18,
      color: 'grey'
    });
  });

  it('throws on a bad schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      color: {
        type: String
      }
    });

    const cat = {
      age: 'eighteen',
      weight: '20 lbs'
    };

    expect(() => schema.validate(cat)).toThrowErrorMatchingSnapshot();
  });
});