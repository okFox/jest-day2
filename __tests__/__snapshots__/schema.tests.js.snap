// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema throws on a bad schema 1`] = `"not a valid schema: Error: name is required,Error: Cannot cast >>eighteen<< to Number"`;
